src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaTransportControls.MediaPlayer.cs

```diff
@@ -123,27 +98,28 @@ private void OnPlaybackStateChanged(MediaPlaybackSession sender, object args)
 				case MediaPlaybackState.None:
 					_mediaPlayer.PlaybackSession.UpdateTimePositionRate = 0;
 					CancelControlsVisibilityTimer();
-					VisualStateManager.GoToState(this, "PlayState", false);
-					VisualStateManager.GoToState(this, "Normal", false);
 					break;
 				case MediaPlaybackState.Playing:
 					_mediaPlayer.PlaybackSession.UpdateTimePositionRate = 0;
 					ResetControlsVisibilityTimer();
-					VisualStateManager.GoToState(this, "PauseState", false);
-					VisualStateManager.GoToState(this, "Normal", false);
 					break;
 				case MediaPlaybackState.Buffering:
-					VisualStateManager.GoToState(this, "Buffering", false);
 					break;
 			}
+
+			// skip transition, as the event may originate from non-ui thread
+			UpdateMediaStates(useTransition: false);
+			if (!_skipPlayPauseStateUpdate)
+			{
+				UpdatePlayPauseStates(useTransition: false);
+			}
 		}
 
 		private void OnBufferingProgressChanged(MediaPlaybackSession sender, object args)
 		{
-
 			_ = Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
 			{
-				_downloadProgressIndicator.Maybe(p => p.Value = (double)args);
+				m_tpDownloadProgressIndicator.Maybe(p => p.Value = (double)args);
```

Comment:
@jeromelaban is the `.Maybe()` extension something we want to avoid in general?

Reply:
yes, it creates a closure that's generally not required, but for this particular class, the performance is not a concern. We can adjust later.